src/Microsoft.AspNetCore.Sockets/ConnectionManager.cs

```diff
@@ -65,38 +78,79 @@ private static void Scan(object state)
 
         private void Scan()
         {
-            // Scan the registered connections looking for ones that have timed out
-            foreach (var c in _connections)
+            if (_running)
             {
-                if (!c.Value.Active && (DateTimeOffset.UtcNow - c.Value.LastSeenUtc).TotalSeconds > 5)
+                return;
+            }
+
+            try
+            {
+                _running = true;
```

Comment:
Will change to do a compare exchange https://github.com/SignalR/SignalR/blob/1fba14fa3437e24c204dfaf8a18db3fce8acad3c/src/Microsoft.AspNet.SignalR.Core/Transports/TransportHeartBeat.cs#L171. 

> is it just in situations where the Scan takes more than one Timer interval? If so, suspend the timer when you start or something.

Yes, you mean `timer.Change(Timeout.Infinite, Timeout.Infinite);`?